pkg/controller/jobs/raycluster/raycluster_controller_test.go

```diff
@@ -0,0 +1,370 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package raycluster
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/podset"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	testingrayutil "sigs.k8s.io/kueue/pkg/util/testingjobs/raycluster"
+)
+
+var (
+	jobCmpOpts = cmp.Options{
+		cmpopts.EquateEmpty(),
+		cmpopts.IgnoreFields(rayv1.RayCluster{}, "TypeMeta", "ObjectMeta"),
+	}
+	workloadCmpOpts = cmp.Options{
+		cmpopts.EquateEmpty(),
+		cmpopts.IgnoreFields(kueue.Workload{}, "TypeMeta", "ObjectMeta"),
+		cmpopts.IgnoreFields(kueue.WorkloadSpec{}, "Priority"),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+		cmpopts.IgnoreFields(kueue.PodSet{}, "Template"),
+	}
+)
+
+func TestReconciler(t *testing.T) {
+	baseJobWrapper := testingrayutil.MakeCluster("job", "ns").
+		Suspend(true).
+		Queue("foo").
+		RequestHead(corev1.ResourceCPU, "10").
+		RequestWorkerGroup(corev1.ResourceCPU, "10")
+
+	cases := map[string]struct {
+		reconcilerOptions []jobframework.Option
+		job               rayv1.RayCluster
+		workloads         []kueue.Workload
+		priorityClasses   []client.Object
+		wantJob           rayv1.RayCluster
+		wantWorkloads     []kueue.Workload
+		runInfo           []podset.PodSetInfo
+		restoreInfo       []podset.PodSetInfo
+		wantErr           error
+	}{
+		"when workload is admitted, cluster is unsuspended": {
+			job: *baseJobWrapper.Clone().
+				NodeSelectorHeadGroup("provisioning", "spot").
```

Comment:
Should this should be removed from the test input, as it's expected to be added during the reconciliation?

Reply:
Also a flavor needs to be defined, because that's where the selectors are obtained from.